contracts/Penalizer.sol

```diff
@@ -23,14 +25,36 @@ contract Penalizer is IPenalizer{
         transaction.value,
         transaction.data) = RLPReader.decodeTransaction(rawTransaction);
         return transaction;
+    }
+
+    mapping(bytes32 => uint) public commits;
 
+    event Commit(bytes32 indexed commitHash, address sender);
+
+    /**
+     * any sender can call "commit(keccak(encodedPenalizeFunction))", to make sure
+     * no-one can front-run it to claim this penalization
+     */
+    function commit(bytes32 msgDataHash, uint revealSalt) external override {
+        bytes32 idx = keccak256(abi.encodePacked(msgDataHash, revealSalt, msg.sender));
```

Comment:
fix: give "idx" as parameter to commit, and no need for revealSalt.
(that is, the hash of (msgDataHash, msgSender) is done off-chain for the commit, but on-chain for the reveal